### PR TITLE
test: bump test timeouts so that functional tests run in valgrind

### PR DESCRIPTION
**Pull request by michiboo**
ci/tests: Bump timeouts so all functional tests run on travis in valgrind #17763

### CODE REVIEW

File: test/functional/feature_block.py [hunk 1]
```diff
@@ -548,7 +548,7 @@ def run_test(self):
         tx.rehash()
         new_txs.append(tx)
         self.update_block(40, new_txs)
-        self.send_blocks([b40], success=False, reject_reason='bad-blk-sigops', reconnect=True)
+        self.send_blocks([b40], success=False, reject_reason='bad-blk-sigops', reconnect=True, timeout=200)
```
Comment (MarcoFalke) [inline test/functional/feature_block.py]: You might just bump the default in `def send_blocks(...`.

However, feature_block.py might be the hardest to get right, as it includes a 1080 block reorg. Not sure how long that will take, but you might want to `export TEST_RUNNER_EXTRA="--exclude feature_block` for now.

File: test/functional/interface_rest.py [hunk 2]
```diff
@@ -285,9 +285,9 @@ def run_test(self):
 
         # Make 3 tx and mine them on node 1
         txs = []
-        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11))
-        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11))
-        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11))
+        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11, timeout=120))
+        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11, timeout=120))
+        txs.append(self.nodes[0].sendtoaddress(not_related_address, 11, timeout=120))
```
Comment (MarcoFalke) [inline test/functional/interface_rest.py]: rpc calls don't have a timeout argument.

Also, this test properly failed with an uninitialized read:

```
 node0 stderr ==27834== Thread 7 b-httpworker.2:

==27834== Conditional jump or move depends on uninitialised value(s)

==27834==    at 0x204563: rest_blockhash_by_height(HTTPRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (rest.cpp:611)

==27834==    by 0x20B4C8: std::_Function_handler<bool (HTTPRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&), bool (*)(HTTPRequest*, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)>::_M_invoke(std::_Any_data const&, HTTPRequest*&&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) (std_function.h:301)

==27834==    by 0x3B90B4: operator() (std_function.h:706)

==27834==    by 0x3B90B4: HTTPWorkItem::operator()() (httpserver.cpp:55)

==27834==    by 0x3B988D: WorkQueue<HTTPClosure>::Run() (httpserver.cpp:114)

==27834==    by 0x3B5968: HTTPWorkQueueRun(WorkQueue<HTTPClosure>*, int) (httpserver.cpp:342)

==27834==    by 0x3BA1CA: __invoke_impl<void, void (*)(WorkQueue<HTTPClosure> *, int), WorkQueue<HTTPClosure> *, int> (invoke.h:60)

==27834==    by 0x3BA1CA: __invoke<void (*)(WorkQueue<HTTPClosure> *, int), WorkQueue<HTTPClosure> *, int> (invoke.h:95)

==27834==    by 0x3BA1CA: _M_invoke<0, 1, 2> (thread:234)

==27834==    by 0x3BA1CA: operator() (thread:243)

==27834==    by 0x3BA1CA: std::thread::_State_impl<std::thread::_Invoker<std::tuple<void (*)(WorkQueue<HTTPClosure>*, int), WorkQueue<HTTPClosure>*, int> > >::_M_run() (thread:186)

==27834==    by 0x662A66E: ??? (in /usr/lib/x86_64-linux-gnu/libstdc++.so.6.0.25)

==27834==    by 0x54876DA: start_thread (pthread_create.c:463)

==27834==    by 0x6FCD88E: clone (clone.S:95)

==27834== 

{
   Temporary suppression of uninitialized read in rest code
   Memcheck:Cond
   fun:_ZL24rest_blockhash_by_heightP11HTTPRequestRKNSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEE
} 
```

Not sure how that could have happened, so you might want to add the suppression for now to the valgrind.supp file

File: ci/test/00_setup_env_native_valgrind.sh [hunk 3]
```diff
@@ -5,11 +5,10 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 export LC_ALL=C.UTF-8
-
+export TEST_RUNNER_EXTRA="--exclude feature_block"
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_native_valgrind.sh]: temporarily, for faster ci runs, you can also `export RUN_UNIT_TESTS=false`

File: test/functional/interface_rest.py (hunk 2 above)
Comment (practicalswift) [reply test/functional/interface_rest.py]: Hrmm, that is super weird. Valgrind false positives are very rare, but I'm unable to reproduce. How the heck was this triggered? :)

File: test/functional/interface_rest.py [hunk 4]
```diff
@@ -164,7 +164,7 @@ def run_test(self):
         # found with or without /checkmempool.
 
         # do a tx and don't sync
-        txid = self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 0.1)
+        txid = self.nodes[0].sendtoaddress(self.nodes[1].getnewaddress(), 0.1, timeout=120)
```
Comment (MarcoFalke) [inline test/functional/interface_rest.py]: As mentioned previously, RPCs don't have timeout argument

**Comment by MarcoFalke** [comment]
Last failure: https://travis-ci.org/bitcoin/bitcoin/jobs/629019510#L3660

Re-run ci, to see if it is reproducible.

File: ci/test/00_setup_env_native_valgrind.sh [hunk 5]
```diff
@@ -5,11 +5,11 @@
...
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_native_valgrind.sh]: If the `feature_abortnode` test is causing issues that are non-trivial to fix, you might exclude it for now:

```suggestion
export TEST_RUNNER_EXTRA="--exclude feature_block,feature_abortnode"
```

File: ci/test/00_setup_env_native_valgrind.sh [hunk 6]
```diff
@@ -5,11 +5,11 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 export LC_ALL=C.UTF-8
-
+export TEST_RUNNER_EXTRA="--exclude feature_block,feature_abortnode,wallet_createwallet"
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_native_valgrind.sh]:
```suggestion
export TEST_RUNNER_EXTRA="--exclude feature_block,rpc_bind.py --ipv4,feature_abortnode,wallet_createwallet"
```

Due to failure here: https://travis-ci.org/bitcoin/bitcoin/jobs/632228754#L2879

File: test/functional/feature_abortnode.py [hunk 7]
```diff
@@ -40,9 +40,9 @@ def run_test(self):
 
             # Check that node0 aborted
             self.log.info("Waiting for crash")
-            wait_until(lambda: self.nodes[0].is_node_stopped(), timeout=60)
+            wait_until(lambda: self.nodes[0].is_node_stopped(), timeout=200)
         self.log.info("Node crashed - now verifying restart fails")
-        self.nodes[0].assert_start_raises_init_error()
+        self.nodes[0].assert_start_raises_init_error(timeout=200)
```
Comment (MarcoFalke) [inline test/functional/feature_abortnode.py]: This does not have a timeout argument. Please reset the change.

**Comment by MarcoFalke** [comment]
To fix the wallet_groups timeout,  you might need a diff that looks like:

```diff
diff --git a/test/functional/wallet_groups.py b/test/functional/wallet_groups.py
index 3cf8aaf3d..f2fa1d3e4 100755
--- a/test/functional/wallet_groups.py
+++ b/test/functional/wallet_groups.py
@@ -16,7 +16,7 @@ class WalletGroupTest(BitcoinTestFramework):
         self.setup_clean_chain = True
         self.num_nodes = 3
         self.extra_args = [[], [], ['-avoidpartialspends']]
-        self.rpc_timeout = 120
+        self.rpc_timeout = 240
 
     def skip_test_if_missing_module(self):
         self.skip_if_no_wallet()

File: ci/test/00_setup_env_native_valgrind.sh [hunk 8]
```diff
@@ -5,11 +5,11 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 export LC_ALL=C.UTF-8
-
+export TEST_RUNNER_EXTRA="--exclude feature_block,rpc_bind,feature_abortnode,wallet_createwallet"
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_native_valgrind.sh]:
```suggestion
export TEST_RUNNER_EXTRA="--exclude feature_block,rpc_bind,feature_abortnode,wallet_createwallet,wallet_multiwallet"
```

Maybe exclude this for now due to the failure  https://travis-ci.org/bitcoin/bitcoin/jobs/636719655#L4607

**Comment by MarcoFalke** [comment]
Looks like feature_assumevalid needs the `self.rpc_timeout = 240` bumped as well. Also, there are about 10 tests left which don't run. You could remove the `--failfast` in `ci/test/06_script_b.sh` temporarily to run them all even if one fails.

**Comment by MarcoFalke** [comment]
Ok, down to three failures :rocket: 

> `feature_assumevalid.py                           | ✖ Failed  | 206 s` 
> `feature_cltv.py                                  | ✖ Failed  | 205 s`

I think they can be excluded for now because they mine too many blocks

> `feature_help.py                                  | ✖ Failed  | 5 s`

Timeout should be bumped here, probably

**Comment by practicalswift** [comment]
Concept ACK

Nice first-time contribution! Welcome as a contributor @michiboo :)

**Comment by MarcoFalke** [comment]
ok, one more timeout on travis. Might want to bump that and remove the temporary workarounds in the ci files `failfast` and `UNIT_TESTS=false`.

Then it should be ready for merge.

**Comment by MarcoFalke** [comment]
Also could adjust the commit title to match the pull request title, which is a bit more descriptive.

**Comment by michiboo** [comment]
Hi There is error: The job exceeded the maximum time limit for jobs, and has been terminated.

Not sure how to avoid this.

File: ci/test/00_setup_env_native_valgrind.sh [hunk 9]
```diff
@@ -5,11 +5,10 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 export LC_ALL=C.UTF-8
-
+export TEST_RUNNER_EXTRA="--exclude feature_block,rpc_bind,feature_assumevalid,feature_cltv,feature_abortnode,wallet_createwallet,wallet_multiwallet"
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_native_valgrind.sh]: ok, then maybe revert the changes to this file altogether, but I'd still like to get this merged. The timeout changes help me running the valgrind tests on my own ci infrastructure. We can look into how to enable them on travis later on.

**Review by maflcko — APPROVED** [review]
ACK. Let's look at travis some time later, but I'd like to get the bumped timeouts in, so that I can start running valgrind on my infrastructure.

File: ci/test/00_setup_env_native_valgrind.sh [hunk 10]
```diff
@@ -5,11 +5,9 @@
 # file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
 export LC_ALL=C.UTF-8
-
 export PACKAGES="valgrind clang llvm python3-zmq libevent-dev bsdmainutils libboost-system-dev libboost-filesystem-dev libboost-chrono-dev libboost-test-dev libboost-thread-dev libdb5.3++-dev libminiupnpc-dev libzmq3-dev"
 export USE_VALGRIND=1
 export NO_DEPENDS=1
-export TEST_RUNNER_EXTRA="p2p_segwit.py"  # Only run one test for now. TODO enable all and bump timeouts
```
Comment (MarcoFalke) [inline ci/test/00_setup_env_native_valgrind.sh]: please reset all changes in this file to restore the inital state of the travis config

**Pull request merged**